### PR TITLE
nmd 490 add business logic

### PR DESCRIPTION
**Pull request by VadimZharikov**
removed exception handling in repositories, added bll items, services, added automapper.

### CODE REVIEW

File: Client/EnlightenmentApp.API/Program.cs [hunk 1]
```diff
@@ -12,6 +12,9 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+builder.Services.AddAutoMapper(
+    typeof(EnlightenmentApp.BLL.Mapper.MappingProfile).Assembly,
```
Comment (Glipchik) [inline Client/EnlightenmentApp.API/Program.cs:16]: Move BLL automapper connect to BLL project

File: Client/EnlightenmentApp.BLL/Services/GenericService.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        public virtual async Task<TItem> GetById(int id, CancellationToken ct)
+        {
+            var result = _mapper.Map<TItem>(await _repository.GetById(id, ct));
+            if (result != null)
+            {
+                return result;
+            }
+
+            throw new KeyNotFoundException();
```
Comment (Glipchik) [inline Client/EnlightenmentApp.BLL/Services/GenericService.cs:37]: Do not throw exception if entity is not found. You can handle it in the services that gonna use simple CRUD services

File: Client/EnlightenmentApp.BLL/Services/GenericService.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            var itemExists = await _repository.EntityExists(id, ct);
+            if (itemExists)
+            {
+                var entity = await _repository.Delete(id, ct);
+                var result = _mapper.Map<TItem>(entity);
+                return result;
+            }
+
+            throw new KeyNotFoundException();
```
Comment (Glipchik) [inline Client/EnlightenmentApp.BLL/Services/GenericService.cs:62]: Make this idempotent

**Review by Glipchik — CHANGES_REQUESTED** [review]

**Review by Genresal — APPROVED** [review]

**Review by Genresal — APPROVED** [review]
Except Gleb comments all LGTM

**Review by Glipchik — APPROVED** [review]

**Pull request merged**